junit-platform-testkit/src/main/java/org/junit/platform/testkit/engine/Events.java

```diff
@@ -259,6 +264,63 @@ public final void assertEventsMatchExactly(Condition<? super Event>... condition
 		assertEventsMatchExactly(this.events, conditions);
 	}
 
+	/**
+	 * Assert that all provided conditions are matched by an {@linkplain Event event}
+	 * contained in this {@code Events} object regardless of order.
+	 * Note that this method does a partial match, i.e. some events may not match any
+	 * of the provided conditions.
+	 *
+	 * <p>Conditions can be imported statically from {@link EventConditions}
+	 * and {@link TestExecutionResultConditions}.
+	 *
+	 * <h4>Example</h4>
+	 *
+	 * <pre class="code">
+	 * executionResults.tests().assertEventsMatchLoosely(
+	 *     event(test("exampleTestMethod"), started()),
+	 *     event(test("exampleTestMethod"), finishedSuccessfully())
+	 * );
+	 * </pre>
+	 *
+	 * @param conditions the conditions to match against; never {@code null}
+	 * @see EventConditions
+	 * @see TestExecutionResultConditions
+	 */
+	@SafeVarargs
+	public final void assertEventsMatchLoosely(Condition<? super Event>... conditions) {
+		Preconditions.notNull(conditions, "conditions must not be null");
+		assertEventsMatchLoosely(this.events, conditions);
+	}
+
+	/**
+	 * Assert that all provided conditions are matched by an {@linkplain Event event}
+	 * contained in this {@code Events} object.
+	 * Note that this method does a partial match, i.e. some events may not match any
+	 * of the provided conditions.
+	 * However, the conditions provided must be in the correct order.
+	 *
+	 * <p>Conditions can be imported statically from {@link EventConditions}
+	 * and {@link TestExecutionResultConditions}.
+	 *
+	 * <h4>Example</h4>
+	 *
+	 * <pre class="code">
+	 * executionResults.tests().assertEventsMatchIncompleteButOrdered(
+	 *     event(test("exampleTestMethod"), started()),
+	 *     event(test("exampleTestMethod"), finishedSuccessfully())
+	 * );
+	 * </pre>
+	 *
+	 * @param conditions the conditions to match against; never {@code null}
+	 * @see EventConditions
+	 * @see TestExecutionResultConditions
+	 */
+	@SafeVarargs
+	public final void assertEventsMatchIncompleteButOrdered(Condition<? super Event>... conditions) {
```

Comment:
If the other method is `assertEventsMatchLoosely`, I think it would make more sense to name this one `assertEventsMatchLooselyInOrder`, `assertEventsMatchLooselyInExactOrder`, `assertEventsMatchLooselyButInOrder`, or similar.

In other words, I think the names should use the same terminology instead of "loosely" vs "incomplete".

Reply:
assertEventsMatchLooselyInOrder would be my favourite

Reply:
done